extensions/blocks/mailchimp/view.js

```diff
@@ -83,6 +84,7 @@ function activateSubscription( block, blogId ) {
 			return;
 		}
 		block.classList.add( 'is-processing' );
```

Comment:
At this point email is validated and the form is hidden, we can remove the event listener of the input